core/src/main/scala/org/ensime/util/FileEditHelper.scala

```diff
@@ -5,13 +5,23 @@ package org.ensime.util
 import java.io.File
 import org.ensime.api._
 
+import scala.tools.refactoring.common.{ NewFileChange, RenameSourceFileChange }
+
 object FileEditHelper {
 
   import scala.tools.refactoring.common.{ Change, TextChange }
 
-  def fromChange(ch: Change): FileEdit = {
+  def fromChange(ch: Change): Seq[FileEdit] = {
     ch match {
-      case ch: TextChange => TextEdit(ch.file.file, ch.from, ch.to, ch.text)
+      case ch: TextChange => Seq(TextEdit(ch.file.file, ch.from, ch.to, ch.text))
+      case nf: NewFileChange => Seq(NewFile(new File(nf.fullName), nf.text))
+      case rf: RenameSourceFileChange =>
+        val sourceFile = rf.sourceFile
+        val newFile = sourceFile.path.replace(sourceFile.name, rf.to)
+        Seq(
+          NewFile(new File(newFile), ""),
+          DeleteFile(new File(sourceFile.path), "")
```

Comment:
we have a convenience constructor for `File` which means you don't need to use the `new` keyword. I think you can also get the underlying `File` of a `SourceFile` by doing `.file.file`